### PR TITLE
refactoring AST results objects

### PR DESCRIPTION
**Pull request by olgakil**

### CODE REVIEW

File: src/main/java/com/checkmarx/sdk/service/scanner/AstScanner.java [hunk 1]
```diff
@@ -33,9 +32,9 @@ protected void applyFilterToResults(AstScaResults combinedResults, ScanParams sc
      */
     @Override
     protected AstScaResults toResults(ResultsBase scanResults) {
-        ASTResults astResults = new ASTResults((AstSastResults) scanResults);
+        ASTResults astResults =  (ASTResults) scanResults;
```
Comment (natalieb07) [inline src/main/java/com/checkmarx/sdk/service/scanner/AstScanner.java:35]: just want to make sure that "new" is removed intentionally

**Review by natalieb07 — APPROVED** [review]

File: src/main/java/com/checkmarx/sdk/config/PropertyFileLoader.java [hunk 2]
```diff
@@ -22,11 +22,11 @@
      *
      * @param filenames list of resource filenames to load properties from.
      *                  If the same property appears several times in the files, the property value from a file will be overridden with the value from the next file.
-     * @throws ASTRuntimeException if no filenames is provided, or if an error occurred while loading a file.
+     * @throws ScannerRuntimeException if no filenames is provided, or if an error occurred while loading a file.
      */
     public PropertyFileLoader(String... filenames) {
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/config/PropertyFileLoader.java]: Looks like the whole PropertyFileLoader class can be deleted (I see it's no longer used).

File: src/main/java/com/checkmarx/sdk/config/RestClientConfig.java [hunk 3]
```diff
@@ -1,7 +1,6 @@
 package com.checkmarx.sdk.config;
 
-import com.checkmarx.sdk.dto.ast.AstConfig;
-import com.checkmarx.sdk.dto.sca.AstScaConfig;
+import com.checkmarx.sdk.dto.sca.ScaConfig;
 import com.checkmarx.sdk.dto.ScannerType;
 import com.checkmarx.sdk.dto.TokenLoginResponse;
 //import com.checkmarx.sdk.scanner.restclient.sast.dto.ReportType;
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/config/RestClientConfig.java]: Please remove the comment.

File: src/main/java/com/checkmarx/sdk/dto/ScanResults.java [hunk 4]
```diff
@@ -182,7 +182,7 @@ public void setOutput(String output) {
     }
 
     public boolean isAstResults(){
-        return Optional.ofNullable(getAstResults()).isPresent() &&  Optional.ofNullable(getAstResults().getResults()).isPresent();
+        return Optional.ofNullable(getAstResults()).isPresent() ;
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/dto/ScanResults.java]: This statement can just be `getAstResults() != null`

File: src/main/java/com/checkmarx/sdk/dto/sca/report/ScaSummaryBaseFormat.java [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 import java.io.Serializable;
 
-public class AstScaSummaryResults implements Serializable {
+public class ScaSummaryBaseFormat implements Serializable {
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/dto/sca/report/ScaSummaryBaseFormat.java:6]: Can the name be just `ScaSummary`?

File: src/main/java/com/checkmarx/sdk/dto/sca/SCAResults.java [hunk 6]
```diff
@@ -18,16 +19,17 @@
...
-    private AstScaSummaryResults summary;
+    private ScaSummaryBaseFormat summaryBaseFormat;
     private String webReportLink;
     private List<Finding> findings;
     private List<Package> packages;
     private boolean scaResultReady;
     private int nonVulnerableLibraries;
     private int vulnerableAndOutdated;
-
+    private Summary summary;
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/dto/sca/SCAResults.java:28]: Why is ast.Summary stored inside sca.SCAResults?

File: src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java [hunk 7]
```diff
@@ -153,7 +141,7 @@ private AstScaConfig getScaSpecificConfig(ScanParams scanParams) {
     protected void validateScanParams(ScanParams scanParams) {
         validateNotNull(scanParams);
 
-        ScaConfig scaConfig = scanParams.getScaConfig();
+        com.checkmarx.sdk.config.ScaConfig scaConfig = scanParams.getScaConfig();
         if (Optional.ofNullable(scaConfig).isPresent()) {
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java]:
```suggestion
        if (scaConfig != null) {
```

**Pull request merged**